multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLoggerProvider.java

```diff
@@ -62,30 +64,61 @@ private String getTaskId(DelegateExecution execution) {
     }
 
     private ProcessLogger createProcessLogger(String spaceId, String correlationId, String activityId, String loggerName, String logName,
-                                              PatternLayout layout) {
-        Logger logger = Logger.getLogger(loggerName);
-        File logFile = getLocalFile(loggerName);
-        logger.removeAllAppenders();
-        logger.addAppender(createAppender(logger.getLevel(), logFile, layout));
-        return new ProcessLogger(logger, logFile, logName, spaceId, correlationId, activityId);
+                                              AbstractStringLayout abstractStringLayout) {
+        File logFile = getLocalFileByLoggerName(loggerName);
+        LoggerContext loggerContext = initializeLoggerContext(loggerName, logFile, abstractStringLayout);
+        Logger logger = loggerContext.getLogger(loggerName);
+        return new ProcessLogger(loggerContext, logger, loggerName, logFile, logName, spaceId, correlationId, activityId);
+    }
+
+    private LoggerContext initializeLoggerContext(String loggerName, File logFile, AbstractStringLayout abstractStringLayout) {
+
+        LoggerContext loggerContext = new LoggerContext(loggerName);
+        FileAppender fileAppender = initializeStartedAppender(loggerContext, logFile, abstractStringLayout, loggerName);
+        loggerContext.getConfiguration()
+                     .addAppender(fileAppender);
+        LoggerConfig loggerConfig = loggerContext.getConfiguration()
+                                                 .getLoggerConfig(loggerName);
+        loggerConfig.setLevel(Level.DEBUG);
+        loggerConfig.addAppender(fileAppender, Level.DEBUG, null);
+        loggerContext.getRootLogger()
+                     .addAppender(fileAppender);
+        disableConsoleLogging(loggerContext);
+        loggerContext.updateLoggers();
+
+        return loggerContext;
     }
 
-    protected File getLocalFile(String loggerName) {
+    protected File getLocalFileByLoggerName(String loggerName) {
         String fileName = loggerName + LOG_FILE_EXTENSION;
         return new File(DEFAULT_LOG_DIR, fileName);
     }
 
-    private Appender createAppender(Level level, File logFile, PatternLayout layout) {
-        FileAppender appender = new FileAppender();
+    private synchronized FileAppender initializeStartedAppender(LoggerContext loggerContext, File logFile, AbstractStringLayout layout,
```

Comment:
This method creates and starts the appender so it basically does 2 things. I'd split it into two methods. And why it is synchronized?